Source/Dafny/Rewriter.cs

```diff
@@ -118,6 +119,36 @@ public abstract class IRewriter {
       Contract.Requires(program != null);
     }
   }
+  // intern project 
+
+  class tobiRewriter : IRewriter {
+    internal override void PostResolve(Program program) {
+      base.PostResolve(program);
+      foreach (var moduleDefinition in program.Modules()) {
+        foreach (var topLevelDecl in moduleDefinition.TopLevelDecls.OfType<TopLevelDeclWithMembers>()) {
+          foreach (var function in topLevelDecl.Members.OfType<Function>()) {
+            if (function.Body is NestedMatchExpr matchExpr) {
+              var idPatterns = matchExpr.Cases.Select(matchCase => matchCase.Pat).OfType<IdPattern>();
+              foreach (var idPattern in idPatterns) {
+                if (char.IsLower(idPattern.Id[0]) == false && idPattern.Arguments == null) {
```

Comment:
Could you extract `idPattern.Arguments == null` into a variable so the name of the variable can indicate why you are doing this check.